llvm/lib/Target/AMDGPU/AMDGPUISelDAGToDAG.cpp

```diff
@@ -1270,8 +1270,8 @@ bool AMDGPUDAGToDAGISel::SelectDSReadWrite2(SDValue Addr, SDValue &Base,
     // (add n0, c0)
     if (isDSOffset2Legal(N0, OffsetValue0, OffsetValue1, Size)) {
       Base = N0;
-      Offset0 = CurDAG->getTargetConstant(OffsetValue0 / Size, DL, MVT::i8);
-      Offset1 = CurDAG->getTargetConstant(OffsetValue1 / Size, DL, MVT::i8);
+      Offset0 = CurDAG->getTargetConstant(OffsetValue0 / Size, DL, MVT::i64);
+      Offset1 = CurDAG->getTargetConstant(OffsetValue1 / Size, DL, MVT::i64);
```

Comment:
Make them i32 instead? We use i32 for most every target constant

Reply:
My thinking is, if we ever use unsigned constants employing all 32 bits, then we would be having the same problem with them. And they are stored as 64-bit ints anyway.